azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/PCFilter.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.management.network;
+
+import com.microsoft.azure.management.apigeneration.Beta;
+import com.microsoft.azure.management.apigeneration.Fluent;
+import com.microsoft.azure.management.network.model.HasProtocol;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.HasParent;
+import com.microsoft.azure.management.resources.fluentcore.model.Attachable;
+import com.microsoft.azure.management.resources.fluentcore.model.HasInner;
+import com.microsoft.azure.management.resources.fluentcore.model.Indexable;
+
+/**
+ * Client-side representation of packet capture filter.
+ */
+@Fluent
+@Beta
+public interface PCFilter extends Indexable,
+        HasParent<PacketCapture>,
+        HasInner<PacketCaptureFilter> {
+    /**
+     * @return protocol to be filtered on.
+     */
+    PcProtocol protocol();
+
+    /**
+     * @return local IP Address to be filtered on. Notation: "127.0.0.1" for single address entry. "127.0.0.1-127.0.0.255" for range. "127.0.0.1;127.0.0.5"? for multiple entries. Multiple ranges not currently supported. Mixing ranges with multiple entries not currently supported. Default = null.
+     */
+    String localIPAddress();
```

Comment:
By reading the possible formats of localIPAddress, it looks like they are not standard IPv4 address, but it can be a range of ips, multiple ip values or single ip value. Similarly for ports (see below). Should we think of exposing another type to wrap ip and port information?

```java
enum IPAddressFormat {
  SINGLE,
  RANGE,
  MULTIPLE
}
enum PortFormat {
  SINGLE,
  RANGE,
  MULTIPLE
}
class PCFilterIPConfiguration {
  IPAddressFormat ipAddressFormat();
  String IPAddress();     // Single IP address when ipAddressFormat() is SINGLE
  String IPAddressRangeStart(); // IP address start range when ipAddressFormat() is RANGE
  String IPAddressRangeEnd(); // IP address end range when ipAddressFormat() is RANGE
  List<String> ipAddresses(); // collection of IP addresses when ipAddressFormat() is MULTIPLE

  PortsFormat portFormat();
  int port();     // Single port when portFormat() is SINGLE
  String portRangeStart(); // IP address start range when portFormat() is RANGE
  String portRangeEnd(); // IP address end range when portFormat() is RANGE
  List<String> ports(); // collection of IP addresses when portFormat() is MULTIPLE
}
```
 
With a model like this we can wrap both localIPAddress() and localPort() getters in a single property

```java
PCFilterIPConfiguration localIPConfiguration();
```

same for remoteIPAddress() and remotePort() 

```java
PCFilterIPConfiguration remoteIPConfiguration();
```